### PR TITLE
Use fake client  for connection pooler

### PR DESCRIPTION
**Pull request by RafiaSabih**
Test for creation and deletion of connection pooler, using k8s fake client API.

### CODE REVIEW

**Comment by FxKu** [comment]
was their a discussion to put it in an extra file an not add it to `connection_pooler_test.go`?

File: pkg/cluster/connection_pooler_test.go [hunk 1]
```diff
@@ -210,133 +251,178 @@ func TestNeedConnectionPooler(t *testing.T) {
...
-					},
+	var cluster = New(
+		Config{
+			OpConfig: config.Config{
+				ConnectionPooler: config.ConnectionPooler{
+					ConnectionPoolerDefaultCPURequest:    "100m",
+					ConnectionPoolerDefaultCPULimit:      "100m",
+					ConnectionPoolerDefaultMemoryRequest: "100Mi",
+					ConnectionPoolerDefaultMemoryLimit:   "100Mi",
+					NumberOfInstances:                    int32ToPointer(1),
```
Comment (sdudoladov) [inline pkg/cluster/connection_pooler_test.go:398]: Why `int32ToPointer(1)` instead of simply `1` ?

Comment (RafiaSabih) [reply pkg/cluster/connection_pooler_test.go:404]: because NumberOfInstances is of type *int32...!

**Comment by sdudoladov** [comment]
:+1:

**Comment by RafiaSabih** [comment]
:+1:

**Pull request merged**